### PR TITLE
task-213-add-rpc-provider-health-check

### PR DESCRIPTION
**Pull request by GreatGaryCooper**
## Check list  

- [x] All Merge Request pipelines should be passed.
- [x] Make sure all the changes are necessary and correct.
- [x] Make sure no needed change missing (all necessary changes are made).
- [x] Merge Request should contains only changes for one purpose (one feature or bug or refactoring).
- [x] All necessary unit tests are written.
- [x] Critical changes should be tested locally (or on staging).

### CODE REVIEW

**Review by WilliamHawkingStephen — APPROVED** [review]

File: src/utils/config/providers.ts [hunk 1]
```diff
@@ -50,6 +50,10 @@ export class Providers {
     for (const provider of this.providers.values()) {
       await provider.startProviders();
     }
+    // Make sure all providers are connected and healthy
+    await Promise.all(
```
Comment (hypersolo) [inline src/utils/config/providers.ts:54]: in getBlockNumber function we get an rpc provider randomly and call getBlockNumber.
by checking health this way, we don't checkk all providers.

**Review by hypersolo — CHANGES_REQUESTED** [review]

File: src/utils/config/providers.ts (hunk 1 above)
Comment (GreatGaryCooper) [reply src/utils/config/providers.ts:87]: Fixed, can you take a look?

File: src/utils/config/providers.ts [hunk 2]
```diff
@@ -1,50 +1,82 @@
...
   private constructor() {
-    // this.providers.set(
-    //   "zksyncv1_goerli",
-    //   ZkSyncV1.getInstance("zksyncv1_goerli", "goerli"),
-    // );
     this.providers.set(
-      "ethereum_goerli",
-      Ethereum.getInstance("ethereum_goerli", "goerli", 5),
+      SupportedNetworks.EthereumGoerli.name,
+      new Evm(
```
Comment (hypersolo) [inline src/utils/config/providers.ts:13]: by calling evm constructor directly, we don't need Ethereum and Arbitrum classes anymore. let's delete them in this PR :love_letter:

File: src/utils/config/providers.ts [hunk 3]
```diff
@@ -53,6 +85,12 @@ export class Providers {
     }
   }
 
+  async pingProviders(): Promise<void> {
+    for (const provider of this.providers.values()) {
```
Comment (hypersolo) [inline src/utils/config/providers.ts:89]: isn't it better to use Promise.all here?

File: src/test/e2e/cancel-order.e2e.test.ts [hunk 4]
```diff
@@ -2,26 +2,33 @@
...
+  let providers: ProvidersManager;
   const math = Utilities.getInstance().math;
   const wallet = ethers.Wallet.createRandom();
   const spiedSetInterval = jest.spyOn(global, "setInterval");
 
   beforeAll(async () => {
     spiedSetInterval.mockImplementation(jest.fn());
-    providers = Providers.getInstance();
+    providers = ProvidersManager.getInstance();
+    const getBlockNumberSpies = providers.getProviders().map((provider) =>
```
Comment (hypersolo) [inline src/test/e2e/cancel-order.e2e.test.ts:25]: do we need these mocks in test files? i mean after we separate ping from start, we don't have to mock getBlockNumber function.

File: src/utils/config/providers.ts (hunk 2 above)
Comment (GreatGaryCooper) [reply src/utils/config/providers.ts:13]: Deleted, thanks 🙏🏻 🌹

File: src/utils/config/providers.ts (hunk 3 above)
Comment (GreatGaryCooper) [reply src/utils/config/providers.ts:89]: Changed, thanks mate 👍🏻

File: src/test/e2e/cancel-order.e2e.test.ts (hunk 4 above)
Comment (GreatGaryCooper) [reply src/test/e2e/cancel-order.e2e.test.ts]: Thanks for the report, removed 👍🏻

File: src/utils/config/providers.ts (hunk 1 above)
Comment (hypersolo) [reply src/utils/config/providers.ts:89]: Thanks buddy :love_you_gesture:

File: src/utils/config/providers.ts (hunk 2 above)
Comment (hypersolo) [reply src/utils/config/providers.ts:13]: :love_letter:

File: src/test/e2e/cancel-order.e2e.test.ts (hunk 4 above)
Comment (hypersolo) [reply src/test/e2e/cancel-order.e2e.test.ts]: noice :cupid:

File: src/utils/config/providers.ts (hunk 3 above)
Comment (hypersolo) [reply src/utils/config/providers.ts]: :heart:

**Review by hypersolo — APPROVED** [review]

**Pull request merged**